iroh/src/endpoint/rtt_actor.rs

```diff
@@ -99,10 +105,10 @@ impl Stream for MappedStream {
                     if !self.was_direct_before && matches!(new_conn_type, ConnectionType::Direct(_))
                     {
                         self.was_direct_before = true;
-                        inc!(MagicsockMetrics, connection_became_direct);
+                        became_direct = true
```

Comment:
Hum, I don't think you want to change this logic.  You can totally become direct more than once for the same connection.

Reply:
I think it is best to keep the refactor faithull to the current behaviour, including all the bugs.  and do the bug fixes in another PR(s).  That will make it much easier to review.

Reply:
I don't think this changes the logic.
Nor do I think the logic had bugs. It executes the body of this `if` only once over the course of `was_direct_before`'s lifetime.

Reply:
My intention was not do fix any bugs or change existing behavior, only to not clone the the metrics once more into each connection type stream, but instead handle the metrics update in the top level actor. So if I changed anything of substance, I will revert this.